packages/muze-legend/src/legend/legend-helper.js

```diff
@@ -27,16 +27,144 @@ export const getScaleInfo = (scale) => {
 };
 
 /**
+ * Function to recompute the stops in array in case to left and right alignment of legend.
  *
- *
- * @param {*} domain
- * @param {*} steps
- *
+ * @param  {string} param - parameter to measure
+ * @param  {Object} requiredMeasure - required measure parameters
+ * @param  {Object} availableMeasure - available meassure parameters
+ * @param  {Array} domainForLegend - array
+ * @param  {Object} smartLabelCalc - smartLabel Manager
+ * @return {Array} - Array
  */
-export const getInterpolatedData = (domain, steps) => {
+const getcomputedArray = (param, requiredMeasure, availableMeasure, domainForLegend, smartLabelCalc) => {
+    // declaring Current and Next Tick Value variable
+    let currentTickValue;
+    let nextTickValue;
+    const upperBound = domainForLegend[domainForLegend.length - 1];
+
+    // calculating pixel required per Tick Values
+    const pixelPerTick = (availableMeasure[param] / upperBound);
+
+    // checking if available max width is smaller than required width for legend.
+    if (availableMeasure[param] >= requiredMeasure[param]) {
+        return domainForLegend;
+    }
+
+    currentTickValue = (smartLabelCalc.getOriSize((domainForLegend[0]))[param]);
+    for (let i = 1; i < domainForLegend.length; i++) {
+        nextTickValue = ((currentTickValue / pixelPerTick) + (currentTickValue / 2) + domainForLegend[i - 1]);
+        if (domainForLegend[i] < nextTickValue) {
+            domainForLegend.splice(i, 1);
+            i -= 1;
+        }
+        currentTickValue = (smartLabelCalc.getOriSize((domainForLegend[i]))[param]);
+    }
+    return domainForLegend;
+};
+
+/**
+ * function to recompute the Stops Array provided to prevent the Overlapping of values
+ * @param  {Array} domainForLegend - Stops Array
+ * @param  {Object} scaleParams - Scale Parameters
+ * @return {Array} - modified Stops Array
+ */
+export const getInterpolatedArrayData = (domainForLegend, scaleParams) => {
+    /* Checking if UpperBound of Domain is Floating or Not.
+     In case of floating constricting it to 2 decimals after point. */
+    let upperBound = domainForLegend[domainForLegend.length - 1];
+    if (!Number.isInteger(upperBound)) {
```

Comment:
1) All Computations after let and const declarations
2) Declare all mutable variables first(let) and then constant ones. Then start the computation of the function